deeppavlov/models/sklearn/sklearn_component.py

```diff
@@ -0,0 +1,357 @@
+from deeppavlov.core.common.log import get_logger
+from deeppavlov.core.models.estimator import Estimator
+
+log = get_logger(__name__)
+
+
+@register("sklearn_component")
+class SklearnComponent(Estimator):
+    """
+    Class implements wrapper for sklearn components for feature extraction,
+    feature selection, classification, regression etc.
+
+    Args:
+        model_name: string with full name of sklearn model to use, e.g. ``sklearn.linear_model:LogisticRegression``
+        save_path: save path for model, e.g. full name ``model_path/model.pkl`` \
+            or prefix ``model_path/model`` (still model will be saved to ``model_path/model.pkl``)
+        load_path: load path for model, e.g. full name ``model_path/model.pkl`` \
+            or prefix ``model_path/model`` (still model will be loaded from ``model_path/model.pkl``)
+        infer_method: string name of class method to use for infering model, \
+            e.g. ``predict``, ``predict_proba``, ``predict_log_proba``, ``transform``
+        kwargs: dictionary with parameters for the sklearn model
+
+    Attributes:
+        model: sklearn model instance
+        model_name: string with full name of sklearn model to use, e.g. ``sklearn.linear_model:LogisticRegression``
+        model_params: dictionary with parameters for the sklearn model without pipe parameters
+        pipe_params: dictionary with parameters for pipe: ``in``, ``out``, ``fit_on``, ``main``, ``name``
+        save_path: save path for model, e.g. full name ``model_path/model.pkl`` \
+            or prefix ``model_path/model`` (still model will be saved to ``model_path/model.pkl``)
+        load_path: load path for model, e.g. full name ``model_path/model.pkl`` \
+            or prefix ``model_path/model`` (still model will be loaded from ``model_path/model.pkl``)
+        infer_method: string name of class method to use for infering model, \
+            e.g. ``predict``, ``predict_proba``, ``predict_log_proba``, ``transform``
+        epochs_done: number of epochs done
+        batches_seen: number of batches seen
+        train_examples_seen: number of train examples seen
+    """
+    def __init__(self, model_name: str,
+                 save_path: Union[str, Path] = None,
+                 load_path: Union[str, Path] = None,
+                 infer_method: str = "predict", **kwargs) -> None:
+        """
+        Initialize component with given parameters
+        """
+
+        super().__init__(save_path=save_path, load_path=load_path, **kwargs)
+        self.model_name = model_name
+        self.model_params = kwargs
+        self.model = None
+        self.pipe_params = {}
+        for required in ["in", "out", "fit_on", "main", "name"]:
+            self.pipe_params[required] = self.model_params.pop(required, None)
+
+        self.load()
+        self.infer_method = infer_method
+        self.epochs_done = 0
+        self.batches_seen = 0
+        self.train_examples_seen = 0
```

Comment:
why do you need `epochs_done` and other logged params here?

Reply:
fixed